components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/internal/SecretManagerComponent.java

```diff
@@ -0,0 +1,133 @@
+import org.wso2.carbon.user.core.service.RealmService;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.DB_TABLE_NAME;
+
+/**
+ * OSGi declarative services component which handles registration and un-registration of configuration management
+ * service.
+ */
+@Component(
+        name = "carbon.secret.mgt.component",
+        immediate = true
+)
+public class SecretManagerComponent {
+
+    private static final Log log = LogFactory.getLog(SecretManagerComponent.class);
+
+    /**
+     * Register SecretManager as an OSGI service.
+     *
+     * @param componentContext OSGI service component context.
+     */
+    @Activate
+    protected void activate(ComponentContext componentContext) {
+
+        BundleContext bundleContext = componentContext.getBundleContext();
+        SecretDAO secretDAO = new SecretDAOImpl();
+        SecretManagerComponentDataHolder.getInstance().setSecretDAOS(new ArrayList<>());
+        bundleContext.registerService(SecretDAO.class.getName(),
+                new CachedBackedSecretDAO(secretDAO), null);
+        bundleContext.registerService(SecretManager.class.getName(),
+                new SecretManagerImpl(), null);
+        bundleContext.registerService(SecretResolveManager.class.getName(),
+                new SecretResolveManagerImpl(), null);
+        SecretManagerComponentDataHolder.getInstance().setSecretManagementEnabled
+                (isSecretManagementEnabled());
+    }
+
+    @Reference(
+            name = "secret.dao",
+            service = org.wso2.carbon.identity.secret.mgt.core.dao.SecretDAO.class,
+            cardinality = ReferenceCardinality.MULTIPLE,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetSecret"
+    )
+    protected void setSecret(SecretDAO secretDAO) {
+
+        if (secretDAO != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Secret DAO is registered in SecretManager service.");
+            }
+
+            SecretManagerComponentDataHolder.getInstance().getSecretDAOS().add(secretDAO);
+            SecretManagerComponentDataHolder.getInstance().getSecretDAOS().sort(Comparator.comparingInt(SecretDAO::getPriority));
+        }
+    }
+
+    protected void unsetSecret(SecretDAO secretDAO) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Purpose DAO is unregistered in SecretManager service.");
+        }
+        SecretManagerComponentDataHolder.getInstance().getSecretDAOS().remove(secretDAO);
+    }
+
+    @Reference(
+            name = "user.realmservice.default",
+            service = RealmService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetRealmService"
+    )
+    protected void setRealmService(RealmService realmService) {
```

Comment:
As we removed the tenant resolving logic, do we need to the RealmService anymore ?